core/src/main/scala/org/apache/spark/ui/SparkUI.scala

```diff
@@ -59,21 +59,25 @@ private[spark] class SparkUI private (
   val killEnabled = sc.map(_.conf.getBoolean("spark.ui.killEnabled", true)).getOrElse(false)
 
 
+  val jobsTab = new JobsTab(this)
+
   val stagesTab = new StagesTab(this)
 
   var appId: String = _
 
   /** Initialize all components of the server. */
   def initialize() {
-    attachTab(new JobsTab(this))
+    attachTab(jobsTab)
```

Comment:
See https://github.com/apache/spark/pull/15603 which reminds me that these two tabs don't need to be members. They can be locals. You can make the change for both here.